### PR TITLE
prevent possible griefing attack when set strategy

### PR DESCRIPTION
**Pull request by ryuheimat**
Fixes https://github.com/code-423n4/2022-01-sandclock-findings/issues/91

### CODE REVIEW

File: contracts/Vault.sol [hunk 1]
```diff
@@ -161,7 +161,7 @@ contract Vault is
             "Vault: invalid vault"
         );
         require(
-            address(strategy) == address(0) || strategy.investedAssets() == 0,
+            address(strategy) == address(0) || strategy.hasAssets() == false,
```
Comment (gabrielpoca) [inline contracts/Vault.sol:164]: What would be the issue if we just dropped the condition instead?

Comment (ryuheimat) [reply contracts/Vault.sol:164]: The tx should be fail if it is unable to set new strategy.

Comment (gabrielpoca) [reply contracts/Vault.sol:164]: Sure, but what I'm asking is, what if we don't check if the strategy has assets?

Comment (ryuheimat) [reply contracts/Vault.sol:164]: the funds in previous strategy will be lost.

**Review by naps62 — APPROVED** [review]

**Pull request merged**